components/homme/src/share/cxx/KernelVariables.hpp

```diff
@@ -28,8 +28,8 @@ struct KernelVariables {
       return 0;
     }
 
-#ifdef KOKKOS_ENABLE_CUDA
-#ifdef __CUDA_ARCH__
+#ifdef HOMMEXX_ENABLE_GPU
+//#ifdef __CUDA_ARCH__
```

Comment:
i disabled this CUDA_ARCH, but should it remain? along with some HIP_ARCH_* ?

Reply:
@ambrad  or @bartgol  -- is cuda_arch needed here? ^^

Reply:
No. But the fix isn't quite right. I recommend this rewrite, getting ride of the "should never happen" impl:
```c++
#ifdef HOMMEXX_ENABLE_GPU
    template <typename ExecSpaceType>
    static KOKKOS_INLINE_FUNCTION typename std::enable_if<
        OnGpu<ExecSpaceType>::value, int>::type
    get_team_idx(const int /*team_size*/, const int league_rank) {
      return league_rank;
    }
#endif
```